File: study/lua/src/app/scene/view/equipmentJade/EquipDesJadeIcon.lua
--
-- Author:liushiyin
-- Date: 2017-02-20 17:35:20
--
local EquipDesJadeIcon = class("EquipDesJadeIcon")
local TypeConvertHelper = require("app.utils.TypeConvertHelper")
local LogicCheckHelper = require("app.utils.LogicCheckHelper")
local EquipJadeHelper = require("app.scene.view.equipmentJade.EquipJadeHelper")

function EquipDesJadeIcon:ctor(target, slot)
    self._target = target
    self._jadeIcon = ccui.Helper:seekNodeByName(self._target, "FileNode_1")
    cc.bind(self._jadeIcon, "CommonJadeIcon")
    self._jadeIcon:setTouchEnabled(false)
    self._textName = ccui.Helper:seekNodeByName(self._target, "TextName")
end

function EquipDesJadeIcon:updateIcon(jadeId)
    if jadeId > 0 then
        self._jadeIcon:updateUI(jadeId)
        local params = self._jadeIcon:getItemParams()
        self._textName:setString(params.name)
        self._textName:setColor(params.icon_color)
        local UIHelper = require("yoka.utils.UIHelper")
        UIHelper.updateTextOutline(self._textName, params.icon_color_outline)
    else
        self._textName:setString(Lang.get("not_inject_in_time"))
        self._textName:setColor(Colors.getColor(5))
    end
end

function EquipDesJadeIcon.instance()
    local file = Path.getCSB("EquipDesJadeIcon", "equipment")
    local nodes = cc.CSLoader:createNodeAndBind(file)
    local rootNode = nodes._resourceNode
    local object = EquipDesJadeIcon.new(rootNode)
    return object, rootNode
end

return EquipDesJadeIcon

File: study/ts/script/scene/view/equipmentJade/EquipDesJadeIcon.ts
import CommonJadeIcon from "../../../ui/component/CommonJadeIcon";
import { Lang } from "../../../lang/Lang";
import { Colors } from "../../../init";
import UIHelper from "../../../utils/UIHelper";

const {ccclass, property} = cc._decorator;

@ccclass
export default class EquipDesJadeIcon extends cc.Component {
    @property({
        type: CommonJadeIcon,
        visible: true
    })
    _jadeIcon: CommonJadeIcon = null;

    @property({
        type: cc.Label,
        visible: true
    })
    _textName: cc.Label = null;

    onLoad() {
        
    }

    onEnable(){
        this._jadeIcon.setTouchEnabled(false);
    }
    updateIcon(jadeId) {
        if (jadeId > 0) {
            this._jadeIcon.updateUI(jadeId);
            var params = this._jadeIcon.getItemParams();
            this._textName.string = (params.name);
            this._textName.node.color = (params.icon_color);
            UIHelper.updateTextOutline(this._textName, params.icon_color_outline);
        } else {
            this._textName.string = (Lang.get('not_inject_in_time'));
            this._textName.node.color = (Colors.getColor(5));
        }
    }
    
}